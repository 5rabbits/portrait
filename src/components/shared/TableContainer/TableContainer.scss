@import '~styles/core';

$filters-width: 320px;
$transition-duration: 200ms;

.TableContainer {
  height: 100%;
  display: flex;
  flex-direction: column;
}

.TableContainer__toolbar {
  flex-shrink: 0;
  width: 100%;
  position: relative;
  z-index: 1;
}

.TableContainer__contents {
  position: relative;
  transition: margin-left $transition-duration ease-out;
  flex: 1;
}

.TableContainer__Container {
  height: 100%;
}

.TableContainer {
  .EmptyView,
  .LoadingScreen {
    position: absolute;
    top: -$table-section-toolbar-height;
    right: 0;
    bottom: 0;
    left: 0;
    height: auto;
  }
}

.TableContainer__toolbar__grid {
  height: $table-section-toolbar-height;
}

.TableContainer__toolbar__filters {
  position: absolute;
}

.TableContainer__toolbar__totals {
  color: $default-color;
  font-size: $font-size-subhead;
  line-height: $line-height-subhead;
  text-align: center;
  margin-left: 100px;
  transition: opacity $transition-duration ease-out;
}

.TableContainer__toolbar__downloads {
  text-align: right;
}

.TableContainer__filters {
  position: fixed;
  left: -$filters-width;
  width: $filters-width;
  transition: left $transition-duration ease-out;
}

.TableContainer__filters-toggle {
  opacity: 1;
}

.TableContainer--filters-visible {
  .TableContainer__toolbar__filters {
    position: fixed;
  }

  .TableContainer__filters {
    left: 0;
  }

  .TableContainer__contents {
    margin-left: $filters-width;
  }

  .TableContainer__filters-toggle {
    opacity: 0;
  }
}

.TableContainer--no-results,
.TableContainer--loading {
  .TableContainer__toolbar__totals,
  .TableContainer__download {
    opacity: 0;
  }
}
