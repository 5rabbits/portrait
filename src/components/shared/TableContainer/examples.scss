.component-example--TableContainer {
  [data-reactroot] {
    width: 100%;
    transform: translate3d(0, 0, 0);
  }
}
