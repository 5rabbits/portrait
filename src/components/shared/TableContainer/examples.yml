code: >
  <TableContainer
    downloadFormat="pdf"
    filters={
      <FiltersForm>
        <div className="form-group">
          <label>Field 1</label>
          <input className="form-control" />
        </div>
      </FiltersForm>
    }
    totals="16 Projects"
    >
    Some contents
  </TableContainer>
