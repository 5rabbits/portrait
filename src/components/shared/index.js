export Container from './Container'
export TableContainer from './TableContainer'
export EmptyBlock from './EmptyBlock'
export EmptyView from './EmptyView'
export Grid from './Grid'
export Loader from './Loader'
export Offscreen from './Offscreen'
export States from './States'
export FiltersForm from './FiltersForm'
