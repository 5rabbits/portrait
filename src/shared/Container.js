import React from 'react'

const Container = props =>
  <div className="container-page">
    {props.children}
  </div>

export default Container
