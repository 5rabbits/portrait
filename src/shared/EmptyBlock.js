import React from 'react'

const EmptyBlock = props =>
  <div className="empty-view">
    <div className="empty-container">
      <div className="empty-dialog">
        {props.children}
      </div>
    </div>
  </div>

export default EmptyBlock
