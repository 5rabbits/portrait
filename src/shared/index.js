export Container from './Container'
export EmptyBlock from './EmptyBlock'
export EmptyView from './EmptyView'
export Grid from './Grid'
export Loader from './Loader'
export Offscreen from './Offscreen'
export States from './States'
export ActionPanel from './ActionPanel'
