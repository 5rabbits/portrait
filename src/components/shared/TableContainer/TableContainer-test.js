import { React, toJSON, mount } from 'test/helper'
import { TableContainer } from 'shared/TableContainer'

describe('TableContainer', () => {
  it('should render with minimum props', () => {
    const component = mount(
      <TableContainer
        onFiltersToggle={() => {}}
        showFilters={false}
        >
        Test
      </TableContainer>,
    )

    expect(toJSON(component)).toMatchSnapshot()
  })

  describe('if props.filters is defined', () => {
    describe('if props.showFilters', () => {
      it('should render the right link', () => {
        const component = mount(
          <TableContainer
            onFiltersToggle={() => {}}
            filters={<div>Some filters</div>}
            showFilters
            >
            Test
          </TableContainer>,
        )

        expect(toJSON(component)).toMatchSnapshot()
      })
    })

    describe('if not props.showFilters', () => {
      it('should render the right link', () => {
        const component = mount(
          <TableContainer
            onFiltersToggle={() => {}}
            filters={<div>Some filters</div>}
            showFilters={false}
            >
            Test
          </TableContainer>,
        )

        expect(toJSON(component)).toMatchSnapshot()
      })
    })
  })

  describe('getDownloadIcon()', () => {
    describe('if props.downloadFormat is csv', () => {
      it('should return a fa-file-excel-o icon', () => {
        const component = mount(
          <TableContainer
            downloadFormat="csv"
            onFiltersToggle={() => {}}
            showFilters={false}
            >
            Test
          </TableContainer>,
        )
        const instance = component.instance()
        const icon = instance.getDownloadIcon()

        expect(icon).toMatchSnapshot()
      })
    })

    describe('if props.downloadFormat is excel', () => {
      it('should return a fa-file-excel-o icon', () => {
        const component = mount(
          <TableContainer
            downloadFormat="excel"
            onFiltersToggle={() => {}}
            showFilters={false}
            >
            Test
          </TableContainer>,
        )
        const instance = component.instance()
        const icon = instance.getDownloadIcon()

        expect(icon).toMatchSnapshot()
      })
    })

    describe('if props.downloadFormat is pdf', () => {
      it('should return a fa-file-pdf-o icon', () => {
        const component = mount(
          <TableContainer
            downloadFormat="pdf"
            onFiltersToggle={() => {}}
            showFilters={false}
            >
            Test
          </TableContainer>,
        )
        const instance = component.instance()
        const icon = instance.getDownloadIcon()

        expect(icon).toMatchSnapshot()
      })
    })

    describe('if props.downloadFormat is word', () => {
      it('should return a fa-file-word-o icon', () => {
        const component = mount(
          <TableContainer
            downloadFormat="word"
            onFiltersToggle={() => {}}
            showFilters={false}
            >
            Test
          </TableContainer>,
        )
        const instance = component.instance()
        const icon = instance.getDownloadIcon()

        expect(icon).toMatchSnapshot()
      })
    })

    describe('if props.downloadFormat is other format', () => {
      it('should return null', () => {
        const component = mount(
          <TableContainer
            downloadFormat={null}
            onFiltersToggle={() => {}}
            showFilters={false}
            >
            Test
          </TableContainer>,
        )
        const instance = component.instance()
        const icon = instance.getDownloadIcon()

        expect(icon).toBeNull()
      })
    })
  })

  describe('filtersRef(filters)', () => {
    describe('if filters is mounting and state.filtersPosition is undefined', () => {
      it('should set the top and bottom position of the filters', () => {
        const component = mount(
          <TableContainer
            onFiltersToggle={() => {}}
            showFilters={false}
            >
            Test
          </TableContainer>,
        )
        const instance = component.instance()
        const filters = {
          getBoundingClientRect: () => ({ top: 20 }),
        }

        component.setState({ filtersPosition: null })
        instance.filtersRef(filters)

        expect(component.state('filtersPosition')).toEqual({
          top: 20,
          bottom: 0,
        })
      })
    })

    describe('if filters is unmounting', () => {
      it('should not change the state', () => {
        const component = mount(
          <TableContainer
            onFiltersToggle={() => {}}
            showFilters={false}
            >
            Test
          </TableContainer>,
        )
        const instance = component.instance()
        const spy = jest.fn()

        component.setState = spy
        instance.filtersRef(null)

        expect(spy).not.toHaveBeenCalled()
      })
    })
  })

  describe('handleDownload(event)', () => {
    it('should prevent the event default action', () => {
      const event = { preventDefault: jest.fn() }
      const component = mount(
        <TableContainer
          onFiltersToggle={() => {}}
          showFilters={false}
          >
          Test
        </TableContainer>,
      )
      const instance = component.instance()

      instance.handleDownload(event)

      expect(event.preventDefault).toHaveBeenCalledTimes(1)
    })

    describe('if props.onDownload is defined', () => {
      it('should call props.onDownload passing the download format', () => {
        const event = { preventDefault: jest.fn() }
        const handleDownload = jest.fn()
        const component = mount(
          <TableContainer
            onFiltersToggle={() => {}}
            showFilters={false}
            onDownload={handleDownload}
            downloadFormat="excel"
            >
            Test
          </TableContainer>,
        )
        const instance = component.instance()

        instance.handleDownload(event)

        expect(handleDownload).toHaveBeenCalledTimes(1)
        expect(handleDownload).toHaveBeenCalledWith('excel')
      })
    })

    describe('if props.onDownload is undefined', () => {
      it('should not fail', () => {
        const event = { preventDefault: jest.fn() }
        const component = mount(
          <TableContainer
            onFiltersToggle={() => {}}
            showFilters={false}
            >
            Test
          </TableContainer>,
        )
        const instance = component.instance()

        expect(() => {
          instance.handleDownload(event)
        }).not.toThrow()
      })
    })
  })

  describe('handleFiltersToggle(event)', () => {
    it('should prevent the event default action', () => {
      const event = { preventDefault: jest.fn() }
      const component = mount(
        <TableContainer
          onFiltersToggle={() => {}}
          showFilters={false}
          >
          Test
        </TableContainer>,
      )
      const instance = component.instance()

      instance.handleFiltersToggle(event)

      expect(event.preventDefault).toHaveBeenCalledTimes(1)
    })

    it('should call props.onFiltersToggle passing the opposite of props.showFilters', () => {
      const event = { preventDefault: jest.fn() }
      const handleFiltersToggle = jest.fn()
      const component = mount(
        <TableContainer
          onFiltersToggle={handleFiltersToggle}
          showFilters={false}
          >
          Test
        </TableContainer>,
      )
      const instance = component.instance()

      instance.handleFiltersToggle(event)

      expect(handleFiltersToggle).toHaveBeenCalledTimes(1)
      expect(handleFiltersToggle).toHaveBeenCalledWith(true)
    })
  })
})
