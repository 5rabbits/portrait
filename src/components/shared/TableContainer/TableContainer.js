import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import { Container, Grid } from 'shared'
import { LayoutLink } from 'controls'
import cx from 'classnames'
import i18n from 'helpers/i18n'
import './TableContainer.scss'

/**
 * Section layout for a table list module.
 */
export default class TableContainer extends PureComponent {
  static propTypes = {
    /**
     * The table to be displayed.
     */
    children: PropTypes.node.isRequired,

    /**
     * CSS class assigned to the container.
     */
    className: PropTypes.string,

    /**
     * Indicates if the filters should start visible or not.
     */
    defaultShowFilters: PropTypes.bool,

    /**
     * The download format available.
     */
    downloadFormat: PropTypes.oneOf([
      'csv', 'excel', 'pdf', 'word',
    ]),

    /**
     * The form component used to filter the list. This should a be a
     * [FiltersForm](/components/FiltersForm) component.
     */
    filters: PropTypes.node,

    /**
     * Callback invoked when the user clicks the download button.
     * The first argument is the selected download format.
     */
    onDownload: PropTypes.func,

    /**
     * Callback invoked when the filters visilibity changes.
     * The first argument is a boolean that indicates if the filters are now visible (`true`) or
     * hidden (`false`).
     */
    onFiltersToggle: PropTypes.func,

    /**
     * Indicates if the filters should be visible or not. Note that this prop is controllable,
     * meaning that you should use `defaultShowFilters` to allow an uncontrolled behaviour.
     */
    showFilters: PropTypes.bool,

    /**
     * Summary information to be displayed on top of the table.
     */
    totals: PropTypes.node,
  }

  static defaultProps = {
    className: null,
    defaultShowFilters: false,
    downloadFormat: null,
    filters: null,
    onFiltersToggle: null,
    onDownload: null,
    showFilters: null,
    totals: null,
  }

  static childContextTypes = {
    toggleFilters: PropTypes.func,
  }

  state = {
    filtersPosition: null,
    showFilters: this.props.showFilters == null
      ? this.props.defaultShowFilters
      : this.props.showFilters,
  }

  getChildContext() {
    return {
      toggleFilters: this.toggleFilters,
    }
  }

  componentWillReceiveProps(nextProps) {
    if (this.props.showFilters !== nextProps.showFilters) {
      this.setState({
        showFilters: nextProps.showFilters,
      })
    }
  }

  getDownloadIcon = () => {
    switch (this.props.downloadFormat) {
      case 'csv':
      case 'excel':
        return <i className="fa fa-file-excel-o" />

      case 'pdf':
        return <i className="fa fa-file-pdf-o" />

      case 'word':
        return <i className="fa fa-file-excel-o" />

      default:
        return null
    }
  }

  toggleFilters = () => {
    const showFilters = !this.state.showFilters

    if (this.props.showFilters == null) {
      this.setState({ showFilters })

      if (!showFilters) {
        document.body.scrollTop = 0
      }
    }

    if (this.props.onFiltersToggle) {
      this.props.onFiltersToggle(showFilters)
    }
  }

  containerRef = container => {
    if (container) {
      this.setState({
        filtersPosition: {
          top: container.getBoundingClientRect().top,
          bottom: 0,
        },
      })
    }
  }

  handleDownload = event => {
    event.preventDefault()

    if (this.props.onDownload) {
      this.props.onDownload(this.props.downloadFormat)
    }
  }

  handleFiltersToggle = () => {
    this.toggleFilters()
  }

  render() {
    const { children, className, downloadFormat, filters, totals, ...other } = this.props
    const { filtersPosition, showFilters } = this.state

    delete other.defaultShowFilters
    delete other.onFiltersToggle
    delete other.onDownload
    delete other.showFilters

    return (
      <div
        {...other}
        className={cx('TableContainer', className, {
          'TableContainer--filters-visible': showFilters,
        })}
        ref={this.containerRef}
        >
        <Container
          className="TableContainer__toolbar"
          size="fluid"
          >
          <Grid
            align="center"
            className="TableContainer__toolbar__grid"
            size={['0 0 100px', 'fill', '0 0 100px']}
            gutter={false}
            >
            <div className="TableContainer__toolbar__filters">
              {filters &&
                <LayoutLink
                  className="TableContainer__filters-toggle"
                  onClick={this.handleFiltersToggle}
                  >
                  <span>
                    {i18n.t('TableContainer.showFilters')}
                    {' '}
                    <i className="fa fa-filter" />
                  </span>
                </LayoutLink>
              }
            </div>
            <div className="TableContainer__toolbar__totals">
              {totals}
            </div>
            <div className="TableContainer__toolbar__downloads">
              {downloadFormat &&
                <LayoutLink onClick={this.handleDownload}>
                  {i18n.t('TableContainer.download')} {this.getDownloadIcon()}
                </LayoutLink>
              }
            </div>
          </Grid>
        </Container>
        {filters &&
          <div
            className="TableContainer__filters"
            ref={this.filtersRef}
            style={filtersPosition}
            >
            {filters}
          </div>
        }
        <div className="TableContainer__contents">
          <Container>
            {children}
          </Container>
        </div>
      </div>
    )
  }
}
