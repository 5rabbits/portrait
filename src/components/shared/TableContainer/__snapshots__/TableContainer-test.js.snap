// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableContainer getContents() if props.loading should display a LoadingScreen 1`] = `
<TableContainer
  className={null}
  defaultShowFilters={false}
  downloadFormat={null}
  filters={null}
  loading={true}
  noResults={false}
  noResultsAction={undefined}
  noResultsIcon={undefined}
  noResultsText={undefined}
  noResultsTitle={undefined}
  onDownload={null}
  onFiltersToggle={null}
  showFilters={null}
  totals={null}
>
  <div
    className="TableContainer TableContainer--loading"
  >
    <Container
      className="TableContainer__toolbar"
      size="fluid"
    >
      <div
        className="Container Container--fluid TableContainer__toolbar"
      >
        <Grid
          align="center"
          className="TableContainer__toolbar__grid"
          direction="horizontal"
          gutter={false}
          justify="start"
          size={
            Array [
              "0 0 100px",
              "fill",
              "0 0 100px",
            ]
          }
        >
          <div
            className="grid-component TableContainer__toolbar__grid grid-component--center-align grid-component--horizontal-direction grid-component--start-justify"
          >
            <div
              className="TableContainer__toolbar__filters"
              key=".0"
              style={
                Object {
                  "flex": "0 0 100px",
                }
              }
            />
            <div
              className="TableContainer__toolbar__totals"
              key=".1"
              style={
                Object {
                  "flex": 1,
                }
              }
            />
            <div
              className="TableContainer__toolbar__downloads"
              key=".2"
              style={
                Object {
                  "flex": "0 0 100px",
                }
              }
            />
          </div>
        </Grid>
      </div>
    </Container>
    <div
      className="TableContainer__contents"
    >
      <Container
        className="TableContainer__Container"
        size="normal"
      >
        <div
          className="Container Container--normal TableContainer__Container"
        >
          <LoadingScreen
            className={null}
            text="Searching"
          >
            <div
              className="LoadingScreen"
            >
              <Spinner
                backgroundColor={null}
                className={null}
                color={null}
                ringSize={4}
                size="5em"
              >
                <div
                  className="Spinner"
                  style={
                    Object {
                      "borderColor": null,
                      "borderLeftColor": null,
                      "borderWidth": 4,
                      "height": "5em",
                      "width": "5em",
                    }
                  }
                />
              </Spinner>
              <div
                className="LoadingScreen__text"
              >
                Searching
              </div>
            </div>
          </LoadingScreen>
        </div>
      </Container>
    </div>
  </div>
</TableContainer>
`;

exports[`TableContainer getContents() if props.noResults should display an EmptyView 1`] = `
<TableContainer
  className={null}
  defaultShowFilters={false}
  downloadFormat={null}
  filters={null}
  loading={false}
  noResults={true}
  noResultsAction={undefined}
  noResultsIcon={undefined}
  noResultsText={undefined}
  noResultsTitle={undefined}
  onDownload={null}
  onFiltersToggle={null}
  showFilters={null}
  totals={null}
>
  <div
    className="TableContainer TableContainer--no-results"
  >
    <Container
      className="TableContainer__toolbar"
      size="fluid"
    >
      <div
        className="Container Container--fluid TableContainer__toolbar"
      >
        <Grid
          align="center"
          className="TableContainer__toolbar__grid"
          direction="horizontal"
          gutter={false}
          justify="start"
          size={
            Array [
              "0 0 100px",
              "fill",
              "0 0 100px",
            ]
          }
        >
          <div
            className="grid-component TableContainer__toolbar__grid grid-component--center-align grid-component--horizontal-direction grid-component--start-justify"
          >
            <div
              className="TableContainer__toolbar__filters"
              key=".0"
              style={
                Object {
                  "flex": "0 0 100px",
                }
              }
            />
            <div
              className="TableContainer__toolbar__totals"
              key=".1"
              style={
                Object {
                  "flex": 1,
                }
              }
            />
            <div
              className="TableContainer__toolbar__downloads"
              key=".2"
              style={
                Object {
                  "flex": "0 0 100px",
                }
              }
            />
          </div>
        </Grid>
      </div>
    </Container>
    <div
      className="TableContainer__contents"
    >
      <Container
        className="TableContainer__Container"
        size="normal"
      >
        <div
          className="Container Container--normal TableContainer__Container"
        >
          <EmptyView
            action={null}
            className={null}
            icon={<Search />}
            text={null}
            title={null}
          >
            <div
              className="EmptyView"
            >
              <div
                className="EmptyView__icon"
              >
                <Search>
                  <svg
                    fill="#000"
                    height="1em"
                    viewBox="0 0 53.6 53.6"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M53.6 49.4L37.8 33.6C40.4 30 42 25.7 42 21 42 9.4 32.6 0 21 0S0 9.4 0 21s9.4 21 21 21c4.7 0 9-1.6 12.6-4.2l15.8 15.8L53.6 49.4zM2 21C2 10.5 10.5 2 21 2s19 8.5 19 19 -8.5 19-19 19S2 31.5 2 21z"
                    />
                  </svg>
                </Search>
              </div>
              <div
                className="EmptyView__title"
              >
                No results
              </div>
              <div
                className="EmptyView__text"
              >
                Try changing the search filters
              </div>
            </div>
          </EmptyView>
        </div>
      </Container>
    </div>
  </div>
</TableContainer>
`;

exports[`TableContainer getDownloadIcon() if props.downloadFormat is csv should return a fa-file-excel-o icon 1`] = `
<i
  className="fa fa-file-excel-o"
/>
`;

exports[`TableContainer getDownloadIcon() if props.downloadFormat is excel should return a fa-file-excel-o icon 1`] = `
<i
  className="fa fa-file-excel-o"
/>
`;

exports[`TableContainer getDownloadIcon() if props.downloadFormat is pdf should return a fa-file-pdf-o icon 1`] = `
<i
  className="fa fa-file-pdf-o"
/>
`;

exports[`TableContainer getDownloadIcon() if props.downloadFormat is word should return a fa-file-word-o icon 1`] = `
<i
  className="fa fa-file-excel-o"
/>
`;

exports[`TableContainer if props.filters is defined should render the toggle filters link 1`] = `
<TableContainer
  className={null}
  defaultShowFilters={false}
  downloadFormat={null}
  filters={
    <div>
      Some filters
    </div>
  }
  loading={false}
  noResults={false}
  noResultsAction={undefined}
  noResultsIcon={undefined}
  noResultsText={undefined}
  noResultsTitle={undefined}
  onDownload={null}
  onFiltersToggle={null}
  showFilters={null}
  totals={null}
>
  <div
    className="TableContainer"
  >
    <Container
      className="TableContainer__toolbar"
      size="fluid"
    >
      <div
        className="Container Container--fluid TableContainer__toolbar"
      >
        <Grid
          align="center"
          className="TableContainer__toolbar__grid"
          direction="horizontal"
          gutter={false}
          justify="start"
          size={
            Array [
              "0 0 100px",
              "fill",
              "0 0 100px",
            ]
          }
        >
          <div
            className="grid-component TableContainer__toolbar__grid grid-component--center-align grid-component--horizontal-direction grid-component--start-justify"
          >
            <div
              className="TableContainer__toolbar__filters"
              key=".0"
              style={
                Object {
                  "flex": "0 0 100px",
                }
              }
            >
              <LayoutLink
                active={false}
                className="TableContainer__filters-toggle"
                disabled={false}
                href="#"
                onClick={[Function]}
              >
                <a
                  className="LayoutLink TableContainer__filters-toggle"
                  href="#"
                  onClick={[Function]}
                >
                  <span>
                    Advanced filters
                     
                    <i
                      className="fa fa-filter"
                    />
                  </span>
                </a>
              </LayoutLink>
            </div>
            <div
              className="TableContainer__toolbar__totals"
              key=".1"
              style={
                Object {
                  "flex": 1,
                }
              }
            />
            <div
              className="TableContainer__toolbar__downloads"
              key=".2"
              style={
                Object {
                  "flex": "0 0 100px",
                }
              }
            />
          </div>
        </Grid>
      </div>
    </Container>
    <div
      className="TableContainer__filters"
      style={
        Object {
          "bottom": 0,
          "top": 0,
        }
      }
    >
      <div>
        Some filters
      </div>
    </div>
    <div
      className="TableContainer__contents"
    >
      <Container
        className="TableContainer__Container"
        size="normal"
      >
        <div
          className="Container Container--normal TableContainer__Container"
        >
          Test
        </div>
      </Container>
    </div>
  </div>
</TableContainer>
`;

exports[`TableContainer should render with minimum props 1`] = `
<TableContainer
  className={null}
  defaultShowFilters={false}
  downloadFormat={null}
  filters={null}
  loading={false}
  noResults={false}
  noResultsAction={undefined}
  noResultsIcon={undefined}
  noResultsText={undefined}
  noResultsTitle={undefined}
  onDownload={null}
  onFiltersToggle={null}
  showFilters={null}
  totals={null}
>
  <div
    className="TableContainer"
  >
    <Container
      className="TableContainer__toolbar"
      size="fluid"
    >
      <div
        className="Container Container--fluid TableContainer__toolbar"
      >
        <Grid
          align="center"
          className="TableContainer__toolbar__grid"
          direction="horizontal"
          gutter={false}
          justify="start"
          size={
            Array [
              "0 0 100px",
              "fill",
              "0 0 100px",
            ]
          }
        >
          <div
            className="grid-component TableContainer__toolbar__grid grid-component--center-align grid-component--horizontal-direction grid-component--start-justify"
          >
            <div
              className="TableContainer__toolbar__filters"
              key=".0"
              style={
                Object {
                  "flex": "0 0 100px",
                }
              }
            />
            <div
              className="TableContainer__toolbar__totals"
              key=".1"
              style={
                Object {
                  "flex": 1,
                }
              }
            />
            <div
              className="TableContainer__toolbar__downloads"
              key=".2"
              style={
                Object {
                  "flex": "0 0 100px",
                }
              }
            />
          </div>
        </Grid>
      </div>
    </Container>
    <div
      className="TableContainer__contents"
    >
      <Container
        className="TableContainer__Container"
        size="normal"
      >
        <div
          className="Container Container--normal TableContainer__Container"
        >
          Test
        </div>
      </Container>
    </div>
  </div>
</TableContainer>
`;
